example/protocol-stack/protocol-stack.cc

```diff
@@ -0,0 +1,62 @@
+/** @file
+
+  an example hello world plugin
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <stdio.h>
+
+#include "ts/ts.h"
+#include "ts/ink_defs.h"
+
+#define DEBUG_TAG "protocol-stack"
+
+static int
+proto_stack_cb(TSCont contp ATS_UNUSED, TSEvent event, void *edata)
+{
+  TSHttpTxn txnp = (TSHttpTxn)edata;
+  char const *results[10];
+  int count = 0;
+  TSDebug(DEBUG_TAG, "Protocols:");
+  TSHttpTxnClientProtocolStackGet(txnp, 10, results, &count);
+  for (int i = 0; i < count; i++) {
+    TSDebug(DEBUG_TAG, "\t%d: %s", i, results[i]);
+  }
+  int ret = TSHttpTxnClientProtocolStackContains(txnp, "h2", NULL);
+  TSDebug(DEBUG_TAG, "Stack %s HTTP/2", ret == 1 ? "contains" : "does not contain");
+  TSHttpTxnReenable(txnp, TS_EVENT_HTTP_CONTINUE);
```

Comment:
Use ``literalinclude`` to include this code sample in the man page (see other examples of this in the docs tree).